bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java

```diff
@@ -154,11 +154,7 @@ public Collection<String> getTypes() {
             return obj;
         }
 
-        Map<String, Object> objects = objectCache.get(scriptIdentifier);
-        if (objects == null) {
-            objects = new HashMap<>();
-            objectCache.put(scriptIdentifier, objects);
-        }
+        Map<String, Object> objects = objectCache.computeIfAbsent(scriptIdentifier, k -> new HashMap<>());
```

Comment:
The problem is that this always creates a warning in the compiler because the EEA can't properly model "return value is non-null if second parameter is non-null" and assumes the return value is nullable.

Unfortunately the ECJ guys don't see the problem or the need to improve that.

Reply:
We had the same situation recently here: https://github.com/openhab/openhab-addons/pull/15828#discussion_r1382012932. Maybe a similar work-around could be used here as well.